File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/speeds/other/MatrixHop.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.features.module.modules.movement.speeds.other;

import net.ccbluex.liquidbounce.LiquidBounce;
import net.ccbluex.liquidbounce.event.*;
import net.ccbluex.liquidbounce.features.module.modules.movement.speeds.SpeedMode;

public class MatrixHop extends SpeedMode implements Listenable {
    public MatrixHop() {
        super("MatrixHop");

        LiquidBounce.eventManager.registerListener(this);
    }

    @Override
    public void onMotion() {
    }

    @Override
    public void onUpdate() {
        if(mc.thePlayer.moveForward > 0) {
            if(mc.thePlayer.onGround) {
                mc.thePlayer.jump();
                mc.thePlayer.setSprinting(true);
                mc.timer.timerSpeed = 0.775F;
                mc.thePlayer.motionX *= 1.08;
                mc.thePlayer.motionZ *= 1.08;
            }else if(mc.thePlayer.fallDistance > 0){
                mc.timer.timerSpeed = 2.5F;
            }
        }
    }

    @Override
    public void onMove(final MoveEvent event) {
    }

    @EventTarget
    public void onMotion(final MotionEvent event) {
    }

    @Override
    public void onEnable() {
        if(mc.thePlayer.onGround)
            mc.thePlayer.motionX = mc.thePlayer.motionZ = 0;
    }

    @Override
    public void onDisable() {
        mc.thePlayer.jumpMovementFactor = 0.02F;
    }

    @Override
    public boolean handleEvents() {
        return isActive();
    }
}

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/speedmodes/matrix/MatrixHop.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.features.module.modules.movement.speedmodes.matrix

import net.ccbluex.liquidbounce.features.module.modules.movement.speedmodes.SpeedMode
import net.ccbluex.liquidbounce.utils.MovementUtils.isMoving
import net.ccbluex.liquidbounce.utils.MovementUtils.strafe
import net.ccbluex.liquidbounce.utils.extensions.tryJump

object MatrixHop : SpeedMode("MatrixHop") {

    override fun onUpdate()  {
        val player = mc.player ?: return
        if (player.isTouchingWater || player.isTouchingLava || player.isInWeb() || player.isClimbing) return

        if (isMoving) {
            if (player.isAirBorne && player.fallDistance > 1.215f) {
                mc.ticker.timerSpeed = 1f
                return
            }

            if (player.onGround) {
                strafe()
                player.tryJump()
                if (player.velocityY > 0) {
                    mc.ticker.timerSpeed = 1.0953f
                }
            } else {
                if (player.velocityY < 0) {
                    mc.ticker.timerSpeed = 0.9185f
                }
            }
        } else {
            mc.ticker.timerSpeed = 1f
        }
    }
}
